java/com/google/flatbuffers/Table.java

```diff
@@ -273,6 +284,320 @@ protected static int compareStrings(int offset_1, byte[] key, ByteBuffer bb) {
     }
     return len_1 - len_2;
   }
+
+  protected int __lookupByBooleanKey(int vectorFieldOffset, int keyFieldOffset, boolean queryKey,
+          boolean defaultValue ) {
+    byte key = (byte)(queryKey ? 1 : 0);
+    byte byteDefaultValue = (byte)(defaultValue ? 1 : 0);
+    return __lookupByByteKey(vectorFieldOffset, keyFieldOffset, key, byteDefaultValue);
+  }
+
+  protected int __lookupByByteKey(int vectorFieldOffset, int keyFieldOffset, int queryKey,
```

Comment:
Hmm.. this is a really terrible level of duplication of code, I think I preferred it as generated code, easier to maintain.